src/tools/run-make-support/src/fs_wrapper.rs

```diff
@@ -0,0 +1,98 @@
+use std::fs;
+use std::path::Path;
+
+/// A safe wrapper around `std::fs::remove_file` that prints the file path if an operation fails.
+pub fn remove_file<P: AsRef<Path>>(path: P) {
```

Comment:
Suggestion: these functions should have `#[track_caller]`, otherwise their panic line info will be inside the support library not in the rmake.rs (I think).